lib/models/s3-event.js

```diff
@@ -0,0 +1,83 @@
+'use strict';
+const S3Event = function (eventData, reqHeaders) {
+  var principal = [];
+  try {
+    const authorization = reqHeaders.authorization.split(':');
+    principal = authorization[0].split(' ');
+  } catch (err) {
+    principal = ['AWS', randomString(21).toUpperCase()];
+  }
+  var eventName = '';
+  var s3Object = {
+    "key": eventData.S3Item.key,
+    "sequencer": ('00' + (Math.floor(Date.now())).toString(16)).toUpperCase()
+  };
+  switch (eventData.eventType) {
+    case 'Copy':
+      eventName = 'ObjectCreated:Copy';
+      s3Object.size = eventData.S3Item.size;
+      break;
+
+    case 'Put':
+      eventName = 'ObjectCreated:Put';
+      s3Object.size = eventData.S3Item.size;
+      s3Object.eTag = eventData.S3Item.md5;
+      break;
+
+    case 'Post':
+      eventName = 'ObjectCreated:Post';
+      s3Object.size = eventData.S3Item.size;
+      s3Object.eTag = eventData.S3Item.md5;
+      break;
+
+    case 'Delete':
+      eventName = 'ObjectRemoved:Delete';
+      break;
+  }
+
+  const event = {
+    "Records": [
+      {
+        "eventVersion": "2.0",
+        "eventSource": "aws:s3",
+        "awsRegion": reqHeaders.host,
+        "eventTime": eventData.S3Item.creationDate || new Date().toISOString(),
+        "eventName": eventName,
+        "userIdentity": {
+          "principalId": 'AWS:' + principal[1],
```

Comment:
This doesn't match the format of the example given in the [documentation](https://docs.aws.amazon.com/AmazonS3/latest/dev/notification-content-structure.html). @anantab do you observe this value prepended with `AWS:` when you get events from real S3?

Reply:
Hi, the real s3 event object has "AWS:" prepended in principalId.